src/utils/did.js

```diff
@@ -76,23 +245,127 @@ export function validateDockDIDSS58Identifier(identifier) {
+ *
+ * --------------------------------------------------------
+ */
+// eslint-disable-next-line no-extend-native
+Object.defineProperty(String.prototype, 'asDid', {
+  get() {
+    if (this.isDid) {
+      return String(this);
+    } else {
+      throw new Error('Not a `Did`');
+    }
+  },
+});
+// eslint-disable-next-line no-extend-native
+Object.defineProperty(String.prototype, 'isDid', {
+  get() {
+    return isHexWithGivenByteSize(String(this), DockDIDByteSize);
+  },
+});
+// eslint-disable-next-line no-extend-native
+Object.defineProperty(String.prototype, 'toStringSS58', {
+  get() {
+    return new DockDid(this.asDid).toStringSS58();
+  },
+});
+
+/**
+ * --------------------------------------------------------
+ */
+
+/**
+ * Takes a DID string, gets the hexadecimal value of that and returns a `DockDidMethodKey` or `DockDid` object.
+ * @param api
+ * @param {string} did -  The DID can be passed as fully qualified DID like `did:dock:<SS58 string>` or
+ * `did:key:<value>` or a 32 byte hex string
+ * @return {string|DockDidOrDidMethodKey} Returns a `string` or `DockDidMethodKey` or `DockDid` object.
+ */
+export function typedHexDID(api, did) {
+  const strDid = did.toString();
+  if (api.specVersion < 50) {
+    return getHexIdentifier(strDid, DockDIDQualifier, DockDIDByteSize);
+  }
+
+  if (strDid.startsWith(DockDidMethodKeySecp256k1Prefix)) {
+    const hex = getHexIdentifier(
+      strDid,
+      DockDidMethodKeySecp256k1Prefix,
+      DockDIDMethodKeySecp256k1ByteSize,
+    );
+
+    return new DockDidMethodKey(new PublicKeySecp256k1(hex));
+  } else if (strDid.startsWith(DockDidMethodKeyEd25519Prefix)) {
+    const hex = getHexIdentifier(
+      strDid,
+      DockDidMethodKeyEd25519Prefix,
+      DockDIDMethodKeyEd25519ByteSize,
+    );
+
+    return new DockDidMethodKey(new PublicKeyEd25519(hex));
+  } else {
+    const hex = getHexIdentifier(strDid, DockDIDQualifier, DockDIDByteSize);
+
+    return new DockDid(hex);
+  }
+}
+
+/**
+ * Gets the hexadecimal value of the given DID received from the substrate side.
+ * @param api
  * @param {string} did -  The DID can be passed as fully qualified DID like `did:dock:<SS58 string>` or
  * a 32 byte hex string
- * @return {string} Returns the hexadecimal representation of the DID.
+ * @return {string|DockDidOrDidMethodKey} Returns an object wrapping the DID.
  */
-export function getHexIdentifierFromDID(did) {
-  return getHexIdentifier(did, DockDIDQualifier, validateDockDIDHexIdentifier, DockDIDByteSize);
+export function typedHexDIDFromSubstrate(api, did) {
+  if (api.specVersion < 50) {
+    return getHexIdentifier(did, DockDIDQualifier, DockDIDByteSize);
+  } else if (did.isDid) {
+    const hex = getHexIdentifier(u8aToHex(did.asDid), [], DockDIDByteSize);
+
+    return new DockDid(hex);
+  } else if (did.isDidMethodKey) {
+    const key = did.asDidMethodKey;
+
+    if (key.isSecp256k1) {
+      const hex = getHexIdentifier(
```

Comment:
As per the implementation of `DockDidMethodKey`, the following 2 calls to it should fail. Is there a test for this branch of code?